### PR TITLE
Return whole response from feature extraction endpoint instead of assuming its shape

### LINKED ISSUE #1647: InferenceClient.feature_extraction makes wrong request to the inference endpoint
### Describe the bug

`InferenceClient.feature_extraction` takes in a `string` argument `text`, but it `POST`s that directly to the inference endpoint instead of wrapping it up in a list.

### Reproduction

_No response_

### Logs

_No response_

### System info

```shell
- huggingface_hub version: 0.16.4
- Platform: macOS-13.4.1-arm64-arm-64bit
- Python version: 3.11.5
- Running in iPython ?: No
- Running in notebook ?: No
- Running in Google Colab ?: No
- Token path ?: /Users/sumit/.cache/huggingface/token
- Has saved token ?: False
- Configured git credential helpers: osxkeychain
- FastAI: N/A
- Tensorflow: N/A
- Torch: N/A
- Jinja2: 3.1.2
- Graphviz: N/A
- Pydot: N/A
- Pillow: N/A
- hf_transfer: N/A
- gradio: N/A
- tensorboard: N/A
- numpy: 1.25.2
- pydantic: 2.3.0
- aiohttp: 3.8.5
- ENDPOINT: https://huggingface.co
- HUGGINGFACE_HUB_CACHE: /Users/sumit/.cache/huggingface/hub
- HUGGINGFACE_ASSETS_CACHE: /Users/sumit/.cache/huggingface/assets
- HF_TOKEN_PATH: /Users/sumit/.cache/huggingface/token
- HF_HUB_OFFLINE: False
- HF_HUB_DISABLE_TELEMETRY: False
- HF_HUB_DISABLE_PROGRESS_BARS: None
- HF_HUB_DISABLE_SYMLINKS_WARNING: False
- HF_HUB_DISABLE_EXPERIMENTAL_WARNING: False
- HF_HUB_DISABLE_IMPLICIT_TOKEN: False
- HF_HUB_ENABLE_HF_TRANSFER: False
```

### PR DESCRIPTION
**Pull request by skulltech**
Fixes #1647

### CODE REVIEW

**Comment by skulltech** [comment]
@Wauplin Please make a release after merging this if possible. I'm using this library in production but this issue came up, so created this quick patch! If released I can install the fix directly through pip.

**Comment by Wauplin** [comment]
Thanks for opening a PR @skulltech . Would you mind adapting the changes as discussed in https://github.com/huggingface/huggingface_hub/issues/1647#issuecomment-1710284284?
Regarding the release, you're lucky ^^ I'm planning a release on either tomorrow or beginning of next week :hugs:

**Comment by skulltech** [comment]
Updated the PR acc to the discussion in the issue. And thanks, would be really happy if it releases that soon!! 😄 @Wauplin

File: tests/test_inference_client.py [hunk 1]
```diff
@@ -136,7 +136,7 @@ def test_document_question_answering(self) -> None:
     def test_feature_extraction(self) -> None:
         embedding = self.client.feature_extraction("Hi, who are you?")
         self.assertIsInstance(embedding, np.ndarray)
-        self.assertEqual(embedding.shape, (8, 768))
+        self.assertEqual(embedding.shape, (1, 8, 768))
```
Comment (osanseviero) [inline tests/test_inference_client.py:139]: Would it be possible to have a second test with a sentence transformers model?

**Review by osanseviero — APPROVED** [review]
LGTM!

File: tests/test_inference_client.py (hunk 1 above)
Comment (Wauplin) [reply tests/test_inference_client.py:139]: Good idea, I just added a small test case in https://github.com/huggingface/huggingface_hub/pull/1648/commits/d9c545d5661abe8aae531d6c06b4d700a5a80830. Should be enough.

**Review by Wauplin — APPROVED** [review]
Looks good to me, thanks for making the change @skulltech !
I've tried it locally and it works as expected.

**Comment by skulltech** [comment]
Haha I just added the test locally and tried to push and then realized you beat me to it @Wauplin 😬 Thanks for the quick work!

**Comment by Wauplin** [comment]
Oooh, sorry about that. I thought you only did changes on Github directly and since adding a test requires to install everything to generate the test cassette itself I thought it'd be easier this way. Anyway, let's ship this now! :)

**Comment by skulltech** [comment]
@Wauplin Haha no worries, thanks! And yes I did originally make the PR through Github but cloned and ran the tests and everything later on. And yes let's ship this 🚀

**Pull request merged**